src/20_Components/amp-autocomplete.html

```diff
@@ -0,0 +1,172 @@
+</head>
+<body>
+  <!-- 
+    ## Basic usage 
+
+    An `amp-autocomplete` must always be nested in a form and have an **input field** and a datasource. 
+    When a user types into this input field, relevant suggestions will automatically appear below the input field.
+
+    A **datasource** must be a JSON object containing an array property `items`, and can be specified inline with a child `script type="application/json"` tag, or a server endpoint specified with a `src` attribute.
+  -->
+  <form>
+    <amp-autocomplete>
+      <input type="text">
+      <script type="application/json">
+        {
+          "items": ["apple", "orange", "banana"]
+        }
+      </script>
+    </amp-autocomplete>
+  </form>
+
+  <!-- 
+    Alternatively, a datasource can also be provided as remote data through the `src` attribute.
+  -->
+  <amp-autocomplete src="<%host%>/json/amp-autocomplete-cities.example.json">
+    <input type="text">
+  </amp-autocomplete>
+
+  <!-- 
+    ## Suggesting rich content
+
+    More complicated data can be passed into autocompleted with an array of JsonObjects in "items".
+
+    ```json
+    {
+      "items": [{
+        "id": 1,
+        "img": "/img/product1_640x426.jpg",
+        "name": "Apple",
+        "price": "1.99",
+        "color": "green"
+      }, 
+      ...
+      }]
+    }
+    ```
+
+    By default, `amp-autocomplete` will suggest items by matching on the "value" property of each JsonObject, but for more specified data, an attribute `filter-value` can be provided to signify the appropriate property to search on. 
+    In the above example, we will want to search on the property `name`, because that is what the user will most likely search for.
+  -->
+  <amp-autocomplete src="<%host%>/json/amp-autocomplete-fruit.json" 
+    filter-value="name">
+    <input type="text">
+    <template type="amp-mustache">
+      <div class="fruit-autocomplete" value="name">
+        <amp-img width="150"
+          height="100"
+          alt="{{name}}"
+          src="{{img}}"></amp-img>
+        <div></div>
+          <div>{{name}}</div>
+          <div>${{price}}</div>
+        </div>
+      </div>
+    </template>
+  </amp-autocomplete>
+
+  <!-- 
+    ## Dynamic `src`
+
+    The content of an `amp-autocomplete` can be changed dynamically by modifying its `src` value using `amp-bind`.
+  -->
+  <div>
+    <div class="suggest-data">
+      <button on="tap:AMP.setState({ srcUrl: '<%host%>/json/amp-autocomplete-countries.example.json' })">Suggest countries</button>
+      <button on="tap:AMP.setState({ srcUrl: '<%host%>/json/amp-autocomplete-cities.example.json' })">Suggest US cities</button>
+    </div>
+    <form method="post"
+      action-xhr="/form/echo-json/post"
```

Comment:
Does this endpoint exist in ABE? The forms should succeed when they are submitted